source/iface/command-accessor.rst

```diff
@@ -6,281 +6,290 @@
-``propList``                `sycl::accessor (buffer) properties`_
-``accessRange``             Dimensions of data to be accessed
-``accessOffset``            Coordinates of origin of data
-==========================  ===
+  template <typename AllocatorT, typename TagT>
+  accessor(buffer<DataT, Dimensions, AllocatorT>& bufferRef,
+           handler& commandGroupHandlerRef, range<Dimensions> accessRange,
+           TagT tag, const property_list& propList = {})
 
-``is_placeholder``
-==================
+  template <typename AllocatorT>
+  accessor(buffer<DataT, Dimensions, AllocatorT>& bufferRef,
+           handler& commandGroupHandlerRef, range<Dimensions> accessRange,
+           id<Dimensions> accessOffset, const property_list& propList = {})
 
-::
+  template <typename AllocatorT, typename TagT>
+  accessor(buffer<DataT, Dimensions, AllocatorT>& bufferRef,
+           handler& commandGroupHandlerRef, range<Dimensions> accessRange,
+           id<Dimensions> accessOffset, TagT tag,
+           const property_list& propList = {})
 
-  constexpr bool is_placeholder() const;
+Constructors of the accessor class.
+The descriptions are listed in |SYCL_ACCESS_CON|.
 
-Return True if this is a placeholder accessor.
+================
+Member functions
+================
 
-``get_size``
-============
+``swap``
+========
 
 ::
 
-  size_t get_size() const;
+  void swap(accessor& other);
 
-Returns size in bytes of the buffer region that this accesses.
+Swaps the contents of the current accessor with
+the contents of ``other``.
 
-``get_count``
-=============
+``is_placeholder``
+==================
 
 ::
 
-  size_t get_count() const;
-
-Returns number elements that this accesses.
-
-``get_range``
-=============
-
-.. parsed-literal::
-
-  *Available only when:
-   dimensions > 0*
-
-  sycl::range<dimensions> get_range() const;
-
-
-.. rubric:: Template parameters
+  bool is_placeholder() const
 
-===============  ===
-``dimensions``   number of dimensions
-===============  ===
-
-Returns dimensions of the associated buffer or range that was
-provided when the accessor was created.
+Returns ``true`` if the accessor is a placeholder.
+Otherwise returns ``false``.
 
 ``get_offset``
 ==============
 
-.. parsed-literal::
-
-  *Available only when:
-   dimensions > 0*
+::
 
-  sycl::id<dimensions> get_offset() const;
+  id<Dimensions> get_offset() const
 
+Available only when ``(Dimensions > 0)``.
 
-.. rubric:: Template parameters
+If this is a ranged accessor, returns the offset
+that was specified when the accessor was constructed.
+For other accessors, returns the default
+constructed ``id<Dimensions>{}``.
 
-===============  ===
-``dimensions``   number of dimensions
-===============  ===
+``get_pointer``
+===============
 
+::
 
-Returns coordinates of the origin of the buffer or offset that was
-provided when the accessor was created.
+  global_ptr<access::decorated::legacy> get_pointer() const noexcept
```

Comment:
Is the other form missing from this list?
`std::add_pointer_t<value_type> get_pointer() const noexcept`

Reply:
Added with https://github.com/KhronosGroup/SYCL_Reference/pull/94/commits/0c470ac494e971cde631e7ae798de38f0b11bdac

Reply:
Thanks.  With the addition as it is now, it's not clear what API is deprecated (there is one line between this API and the previous one, saying "deprecated in 2020. ..."

Can this please be clarified?  It needs to be clear which of the APIs is/are deprecated.

Reply:
Updated with commit https://github.com/KhronosGroup/SYCL_Reference/pull/94/commits/d94161d3a43c761647f2c1926137140abcf42255

Reply:
Which form of `get_pointer` the comment refers to still isn't clear.  "This version" could refer to the version of `get_pointer` either on the line preceding or following.  "The version" could be changed to either "The previous version" or "The following version" to clarify.

![image](https://github.com/KhronosGroup/SYCL_Reference/assets/34314961/833068a0-40f5-41a4-a136-949ea6f7ac8d)

Reply:
I totally agree...added "The previous version" with commit : https://github.com/KhronosGroup/SYCL_Reference/pull/94/commits/aa2f21f4449fd57c51843413d927c72559bbf108